src/displayBackend/wayland/Surface.cpp

```diff
@@ -118,6 +142,44 @@ void Surface::clear()
 /*******************************************************************************
  * Private
  ******************************************************************************/
+#ifdef WITH_WAYLAND_PRESENTATION_API
+void Surface::sPresentationFeedbackHandleSyncOutput(void *data,
+	struct wp_presentation_feedback *feedback, struct wl_output *output) {
+	/* Not interested */
```

Comment:
Don't you need to destroy feedback in this case?

Reply:
As presentation can be synchronized to only one output at a
        time, this event tells which output it was. This event is only
        sent prior to the presented event.

        As clients may bind to the same global wl_output multiple
        times, this event is sent for each bound instance that matches
        the synchronized output. If a client has not bound to the
        right wl_output global at all, this event is not sent.

Reply:
I don't get it. If you have `sPresentationFeedbackHandleSyncOutput` implemented, then you expect it to be called, right?

In `sPresentationFeedbackHandlePresented` and in `sPresentationFeedbackHandleDiscarded` you are calling `wp_presentation_feedback_destroy`. So, why `sPresentationFeedbackHandleSyncOutput` is different?

Reply:
Because this event will be sent every time prior to sPresentationFeedbackHandlePresented per one presentation. So in most implementation examples, it is left unused. But  the most confusing part is that it looks like there is no need manually free feedback_event at all since the protocol states:
"Once a presentation_feedback object has delivered a 'presented' or 'discarded' event it is automatically destroyed." I will retest this and update the patch.